### PR TITLE
[ramda] chore: update jsdoc

### PR DESCRIPTION
**Pull request by somebody1234**
Some notes:
- Quite a few docs have been changed quite significantly from Ramda docs. Feel free to look through them and see if any are worth upstreaming
- All functions were tested with TypeScript at some point. They may not work with current typings, however I think it's fine for now since I'm planning to make quite a few changes to typings
- Some functions have examples that do not work in TypeScript; for those the examples have not been added for now
- Other functions required quite significant changes to examples to get them to work. Some required removing lines completely too
  - One major change was changing ES5-style function classes to ES6 `class`es.
- Most functions have had newlines added in the comment even if the doc has not been changed much. I don't think this is very important given that most maintainers probably have line wrapping and/or wide editor buffers, but just a heads up
- I've also removed `tslint:disable:max-line-length` while I was at it... shouldn't hurt since it's no longer an issue after the reformat a while back.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: docs updated from https://ramdajs.com/docs/

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ramda ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ramda@latest/))
was missing the following properties:
1. collectBy
1. modify
1. modifyPath
1. partialObject
1. on

<details>
<summary>as well as these 5 other properties...</summary>
<p>promap, sequence, splitWhenever, unwind, whereAny</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 97981ebd2cfc19df84d9965d5af9d4fcd7cbade3
</p>

**Review by Nicholaiii — APPROVED** [review]
Huge contribution, thanks!

File: types/ramda/index.d.ts [hunk 1]
```diff
@@ -214,7 +326,7 @@ export function any<T>(fn: (a: T) => boolean): (list: readonly T[]) => boolean;
  * ```typescript
  * const isClub = R.propEq('suit', '♣');
  * const isSpade = R.propEq('suit', '♠');
- * const isBlackCard = R.anyPass([isClub, isSpade]);
+ * const isBlackCard = R.anyPass<Card>([isClub, isSpade]);
```
Comment (adispring) [inline types/ramda/index.d.ts:329]: Should we add `Card` type/interface ?

**Review by adispring — CHANGES_REQUESTED** [review]

File: types/ramda/index.d.ts [hunk 2]
```diff
@@ -1015,33 +1661,97 @@ export function eqBy<T>(fn: (a: T) => unknown): {
...
+ * Dispatches symmetrically to the `equals` methods of both arguments, if present.
+ *
+ * @example
+ * ```typescript
+ * R.equals(1, 1); //=> true
+ * R.equals([1, 2, 3], [1, 2, 3]); //=> true
+ *
+ * type Recursive = { v: Recursive; };
+ *
+ * const a: Recursive = {} as never; a.v = a;
```
Comment (adispring) [inline types/ramda/index.d.ts:1692]: What's the intension to assert `{}` to `never`

File: types/ramda/index.d.ts (hunk 1 above)
Comment (somebody1234) [reply types/ramda/index.d.ts:329]: yeah, nice catch, thanks!

File: types/ramda/index.d.ts (hunk 2 above)
Comment (somebody1234) [reply types/ramda/index.d.ts:1692]: sorry, habit. i'm assuming `as Recursive` will be a lot better?
(i think there's no way to directly construct a `Recursive`)

Comment (somebody1234) [reply types/ramda/index.d.ts:1692]: the reason for the habit: it's a lot easier than typing `as <complicated type>` when the type is complicated, and a tiny bit (but not much) safer than `as any`

File: types/ramda/index.d.ts (hunk 1 above)
Comment (somebody1234) [reply types/ramda/index.d.ts]: just realized it doesn't quite work right now. in the other PR i had changed the type parameters to something i *think* is a bit more ergonomic, but might as well have examples that actually work for now. also made `anyPass` and `allPass` consistent

**Review by adispring — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

**Comment by somebody1234** [comment]
Ready to merge

**Pull request merged**